### PR TITLE
Fix gravity for areas in space near station

### LINKED ISSUE #76827: Solar panel tiles have gravity
https://github.com/tgstation/tgstation/assets/3625094/354f7413-1e5e-4dfe-bb03-dbae5f14ce06

Tiles under the solar panels are magnetic.

This was fixed a few months ago, but seems to happen again.

## Reproduction:

1. Try walking onto a solar panel tile
2. Notice that you lose weightless status

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request
- Fixes #76827

This removes gravity from the following areas:
- Solars
- Nearstation
- Space
- Ordnance bomb testing

##### Note - Asteroid areas (Tram) or planet areas (Ice) are not affected. 

<details>
<summary>Examples of outside gravity turfs</summary>

dreamseeker_Mml3HmRaIK

</details>

Now stepping on floor plating in those areas will result in mobs drifting. (like they do inside the station when gravity is offline) Stepping on or near catwalks, lattices, or walls stills controls your movement. Also refactored a little bit of the `has_gravity` code to use defines instead of `TRUE/FALSE`. 

## Why It's Good For The Game
Consistency. Mag boots are highly recommended when performing EVA.

## Changelog
:cl:
fix: Fix gravity for areas in space near station (solars, nearspace, bomb testing, etc.)
/:cl:

### CODE REVIEW

**Comment by CRITAWAKETS** [comment]
fairly certain this is intended and not a bug
the gravity generator is what is causing gravity to work on tiles in space, rather than being area-based.

File: code/__DEFINES/gravity.dm [hunk 1]
```diff
@@ -47,7 +47,7 @@
  * This should only be possible on multi-z maps because it works like shit on maps that aren't.
  */
 #define NEGATIVE_GRAVITY -1
-
+// #define NONE 0 Used to indicate no gravity
```
Comment (MrMelbert) [inline code/__DEFINES/gravity.dm:50]: Just make a new define `NO_GRAVITY`, I don't like using `NONE` for non-bitlfags

File: code/game/area/areas/station/science.dm [hunk 2]
```diff
@@ -126,4 +126,7 @@
 /area/station/science/ordnance/bomb
 	name = "\improper Ordnance Bomb Site"
 	icon_state = "ord_boom"
+	area_flags = BLOBS_ALLOWED | UNIQUE_AREA | CULT_PERMITTED | NO_GRAVITY
+
+/area/station/science/ordnance/bomb/planet
 	area_flags = BLOBS_ALLOWED | UNIQUE_AREA | CULT_PERMITTED
```
Comment (MrMelbert) [inline code/game/area/areas/station/science.dm:132]: `parent_type::area_flags &= ~NO_GRAVITY` perhaps?

File: code/__DEFINES/gravity.dm (hunk 1 above)
Comment (timothymtorres) [reply code/__DEFINES/gravity.dm]: https://github.com/tgstation/tgstation/pull/88176/commits/57393265eb84630e9f611c3094ee6c900d1ed767

**Comment by timothymtorres** [comment]
Alright we gucci now. Two minor details I should point out:

## Tram Solars
The tram solar wires became orphaned when I split the outside/inside solars area. This caused the unit test to fail for the power grid datum checks. To resolve it I made sure the wire network actually connected to the maint module wiring instead of just halfway. The solar panels themselves are still disconnected and haven't been touched.

## Ordnance bomb testing
I launched some TTVs using the mass driver at the bomb testing site to see if zero gravity would result in the bombs bouncing and then drifting back to sender. So far they seem to bounce at different angles and remain at the bomb site.

**Pull request merged**